arrows/serialize/json/register_algorithms.cxx

```diff
@@ -66,6 +67,9 @@ register_factories( kwiver::vital::plugin_loader& vpm )
   reg.register_algorithm< string >( "kwiver:video_name" );
 
   reg.mark_module_as_loaded();
+
+  // register the algo which writes metadata maps to disk
+  vpm.ADD_ALGORITHM("json", serialize_metadata);
```

Comment:
It seems a little odd that an algorithm is being added this way, although it is supported. Maybe move this above the `mark_module_as_loaded()` call.

Reply:
Additionally, this differs from uses elsewhere in that it does not set attributes on the algorithm.

Reply:
I moved this up as suggested. I tried to do this the same way as the other algorithms but this resulted in the implementation type changing to `serialize-json`, which seems unhelpful. See the output from `plugin_explorer` below:
```
Info on algorithm type "serialize-json" implementation "json_serialize_metadata_map"
  Plugin name: json_serialize_metadata_map
      Perform IO on video metadata using JSON.
```

I'm happy to look further into this if that's useful, but I know very little about the registration process.

Reply:
The naming convention for serializers is formalized to support the serialization and deserialization processes which select the appropriate algorithm based on the sprokit semantic type name and the type of serialization to be done.